GeoLib/AnalyticalGeometry-impl.h

```diff
@@ -118,17 +118,31 @@ void computeRotationMatrixToXY(MathLib::Vector3 const& n,
         T_MATRIX & rot_mat)
 {
     // check if normal points already in the right direction
-    if (sqrt(n[0]*n[0]+n[1]*n[1]) == 0) {
-        rot_mat(0,0) = 1.0;
-        rot_mat(0,1) = 0.0;
-        rot_mat(0,2) = 0.0;
-        rot_mat(1,0) = 0.0;
-        rot_mat(1,1) = 1.0;
-        rot_mat(1,2) = 0.0;
-        rot_mat(2,0) = 0.0;
-        rot_mat(2,1) = 0.0;
-        rot_mat(2,2) = 1.0;
-        return;
+    if (n[0] == 0 && n[1] == 0) {
+        if (n[2] > 0) {
+            rot_mat(0,0) = 1.0;
+            rot_mat(0,1) = 0.0;
+            rot_mat(0,2) = 0.0;
+            rot_mat(1,0) = 0.0;
+            rot_mat(1,1) = 1.0;
+            rot_mat(1,2) = 0.0;
+            rot_mat(2,0) = 0.0;
+            rot_mat(2,1) = 0.0;
+            rot_mat(2,2) = 1.0;
+            return;
+        } else {
+            // rotate by pi about the y-axis
+            rot_mat(0,0) = -1.0;
+            rot_mat(0,1) = 0.0;
+            rot_mat(0,2) = 0.0;
+            rot_mat(1,0) = 0.0;
+            rot_mat(1,1) = 1.0;
+            rot_mat(1,2) = 0.0;
+            rot_mat(2,0) = 0.0;
+            rot_mat(2,1) = 0.0;
+            rot_mat(2,2) = -1.0;
+            return;
+        }
```

Comment:
Why not modify the first and third diagonal entries via: `rot_mat(i,i) = n[2]>0 ? 1.0 : -1.0;` (i = 0,2) and not duplicate the setting of the other matrix entries?